### PR TITLE
gpsd: update to 3.25, with earlier versions included

### PR DESCRIPTION
**Pull request by fhgwright**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

Mac OS X 10.5.8 9L31a, PPC, Xcode 3.1.4 9M2809
Mac OS X 10.5.8 9L31a, i386, Xcode 3.1.4 9M2809
Mac OS X 10.5.8 9L31a, x86_64, Xcode 3.1.4 9M2809
Mac OS X 10.6.8 10K549, i386, Xcode 3.2.6 10M2518
Mac OS X 10.6.8 10K549, x86_64, Xcode 3.2.6 10M2518
Mac OS X 10.7.5 11G63, x86_64, Xcode 4.6.3 4H1503
OS X 10.8.5 12F2560, x86_64, Xcode 5.1.1 5B1008
OS X 10.9.5 13F1911, x86_64, Xcode 6.2 6C131e
OS X 10.10.5 14F2511, x86_64, Xcode 7.2 7C68
OS X 10.11.6 15G22010, x86_64, Xcode 8.1 8B62
macOS 10.12.6 16G2136, x86_64, Xcode 9.2 9C40b
macOS 10.13.6 17G14042, x86_64, Xcode 10.1 10B61
macOS 10.14.6 18G9323, x86_64, Xcode 11.3.1 11C505
macOS 10.15.7 19H2, x86_64, Xcode 12.4 12D4e
macOS 11.7.3 20G1116, arm64, Xcode 13.2.1 13C100
macOS 12.6.3 21G419, arm64, Xcode 14.2 14C18
macOS 13.2 22D49, arm64, Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [NOTE] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [N/A] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

NOTE:
This PR includes commits for earlier versions as well, so that anyone who really wants such a version can just `git checkout` the appropriate Portfile.  This costs very little, since the buildbots and CI only see the latest version.  All versions were fully tested.

### CODE REVIEW

**Comment by fhgwright** [comment]
This is actually finished, but it shouldn't be merged until after #17519 is merged, to avoid breaking currently working 10.5 cases (due to adding the missing `asciidoctor` dependency).

Also, #17519 (nomaintainer) is waiting on #17386 (_way_ past the maintainer timeout).

**Review by michaelld — APPROVED** [review]
LGTM ... let's merge in the order requested

**Pull request merged**
